File: SMART/app/src/main/java/com/cs565project/smart/db/Converters.java
package com.cs565project.smart.db;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

/**
 * Converters required for Room database.
 */

public class Converters {
    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: SMART/app/src/main/java/com/cs565project/smart/db/Converters.kt
package com.cs565project.smart.db

import android.arch.persistence.room.TypeConverter
import java.util.*

/**
 * Converters required for Room database.
 */

class Converters {
    @TypeConverter
    fun fromTimestamp(value: Long?): Date? {
        return if (value == null) null else Date(value)
    }

    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? {
        return date?.time
    }
}
